#1 - Fix some CSS issues on mobile screen

SebinSong opened the pull request:
closes #1 

**[ From ]**

<img src='https://github.com/okTurtles/group-income-simulator/assets/17641213/b00a8f25-d17a-4ead-a5f2-93f96fe54be1' width='380'>

**[ To ]**

<img src='https://github.com/okTurtles/group-income-simulator/assets/17641213/2ec1c6c2-66d9-4492-8ffe-29b3934ee9f2' width='380'>

---
**[ Summary of work ]**

As you can see from the screenshot attached, I actually worked on reducing the size of the footer as well, as part of fixing the problem (Thought there's no need for it too take that large space of the mobile screen). 

Also, the original app paints the content in a confined space (which is 100% viewport height - toolbar - footer) and that forces the banner UI to become extremely small as the group size become larger. This actually made me think, "why not using scrollbar?". So I added a fix where the app uses scrollbar when the content area start to become crowded.

SebinSong committed:
remove RealTimeSimulator.html which is a duplicate of index.html

SebinSong committed:
re-design / fix the bug in the header

SebinSong committed:
some improvements on the existing logics / styles

SebinSong committed:
add .no-img style to the banners

SebinSong committed:
some style updates to the footer

SebinSong committed:
media-query for the footer so the content has larger space

SebinSong committed:
media-query for the toolbar reset-btn

SebinSong committed:
treeshaking the javascript codes for the future

SebinSong committed:
remove Lato from the project

SebinSong committed:
add a logic to determine 'too-tall' column size

SebinSong committed:
improve options tooltip UI

SebinSong committed:
remove .no-img class / improve the presentation in small screen size

SebinSong committed:
fix the bug where the scrollY position is not taken into account

SebinSong committed:
fix the options menu not being cleared when reset

SebinSong commented:
@taoeffect 

There is actually a lot more things to improve in this project. (Which I didn't get to touch as that might lead to causing another bug)

**Firstly,** The entire app is actually rendered as animation if you see **[this line](https://github.com/okTurtles/group-income-simulator/blob/d71344cdf13ec2b5524c90654906ba725ed5210e/index.html#L827)** of the original code, even when there is no visible animation going in the content.
My guess is that, Ian originally had some webGL animation for this app and then removed it later on but then didn't get to cleanly delete all the animation-related code as it's too giant. This should be entirely removed.

**Secondly,** I kept wishing this app was a Vue app while working on this source-code.
This app is essentially an app whose UI updates are driven by the state changes (e.g. group-size changes -> more banners are displayed.). Also, there is a lot of UI elements that are created as a raw html string with a bunch of inline styles and then injected to the DOM(e.g. [here](https://github.com/okTurtles/group-income-simulator/blob/d71344cdf13ec2b5524c90654906ba725ed5210e/index.html#L2340)).
I wished they were a reusable Vue component. that would make it a lot easier to maintain this app.

**Thirdly,** tooltips are rendered inside a separate layer from the UI component that triggers it.
Normally tooltip for a certain component exists either as a sibling or a parent element because it's easier to show/hide it that way. But what this app does 
 when mouse-click happens is, 1) check mouse x, y coordinate  2) check if there's something underneath that position 3) decide/create the content of the tooltip on the fly and inject it the separate layer using `position:absolute` and `top: n`, `y: n`.
 This makes it very really hard to maintain them.
 
**Last but not least,**
there is no SASS and JS module system.

and so on...

---

For better further development, I think converting this entire source-code into 'AstroJS/Vue' project would help.  If you agree with these too, pls create an issue and assign me to it. I can work on it in the future when GroupIncome project becomes less busy.

Cheers,

taoeffect commented on RealTimeSimulator.html:
```diff
@@ -1,2682 +0,0 @@
-<!DOCTYPE html>
-<html>
-
-<head>
-  <title>Group Income Simulator</title>
-  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no" />
-  <link rel="icon" type="image/svg+xml" href="favicon.ico">
-  <link rel="stylesheet" href="style.css">
-</head>
-
-<body>
-  <div class="gis_layout">
```
This file is very important as it contains the version of the simulator that demonstrates the real-time algorithm (that we never ended up using). We definitely do not want to delete it.

SebinSong replied:
Ok, will add it back then. (noticed this is used in nowhere so deleted)

taoeffect commented on fonts/Lato/OFL.txt:
```diff
@@ -1,93 +0,0 @@
-Copyright (c) 2010-2014 by tyPoland Lukasz Dziedzic (team@latofonts.com) with Reserved Font Name "Lato"
```
Why were all of these fonts deleted? Weren't they being used?

SebinSong replied:
@taoeffect `Lato` was in use but in fewer places. (e.g. only in buttons and some input element)
I realised `Poppin` is the font used for input elements and buttons in both Group Income and the new `groupincome.org` website projects, so thought we would do the same in this project too.

taoeffect commented on index.html:
```diff
@@ -6,6 +6,11 @@
   <link rel="icon" type="image/svg+xml" href="favicon.ico">
   <link rel="stylesheet" href="style.css">
+
+  <!-- 
+    Rest In Peace Ian, a.k.a CrazedCoding (https://github.com/CrazedCoding),
+    who was the original author of this project and a truly talented front-end engineer.
+  -->
```
🥲🙇‍♂️🙏

taoeffect reviewed (changes requested):
Great work @SebinSong!

I left a few comments, and also found a few issues.

Here's what this PR looks like with 3v3:

![Screenshot 2024-06-12 at 9 43 23 AM](https://github.com/okTurtles/group-income-simulator/assets/138706/a6c119b5-ecd1-448d-ab01-7b8053b4cd9f)

Here's what it's supposed to look like:

![Screenshot 2024-06-12 at 9 43 27 AM](https://github.com/okTurtles/group-income-simulator/assets/138706/e6f7bee0-6095-4ff4-a684-cbdc7316d90c)

Perhaps related to this, but the positioning on the boxes when users are added is also off in this PR.

I do like the new positioning on the new "Add Pledger" and "Add Receiver" buttons on mobile, but prefer the original blue colors on the buttons as well as the background, can those be reverted?

> For better further development, I think converting this entire source-code into 'AstroJS/Vue' project would help. If you agree with these too, pls create an issue and assign me to it. I can work on it in the future when GroupIncome project becomes less busy.

I do agree, but also think it's fairly low priority.

SebinSong committed:
add RealTimeSimulator.html back in

SebinSong committed:
revert back the footer colouring

SebinSong committed:
fix the graphic distorted bug

SebinSong commented:
@taoeffect 
Fixed the bug you discovered and also reverted the footer colors.

![image](https://github.com/okTurtles/group-income-simulator/assets/17641213/ba4a2ca4-8a45-4c06-8800-c4ba938b4508)

SebinSong committed:
fix header spacing

SebinSong committed:
add some spacing to distribution-locked msg
